src/Servers/Connections.Abstractions/ref/Microsoft.AspNetCore.Connections.Abstractions.netcoreapp.cs

```diff
@@ -123,8 +129,44 @@ public partial interface IConnectionListenerFactory
     {
         System.Threading.Tasks.ValueTask<Microsoft.AspNetCore.Connections.IConnectionListener> BindAsync(System.Net.EndPoint endpoint, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken));
     }
-    public partial interface IMultiplexedConnectionListenerFactory : Microsoft.AspNetCore.Connections.IConnectionListenerFactory
+    public partial interface IMultiplexedConnectionBuilder
     {
+        System.IServiceProvider ApplicationServices { get; }
+        Microsoft.AspNetCore.Connections.MultiplexedConnectionDelegate Build();
+        Microsoft.AspNetCore.Connections.IMultiplexedConnectionBuilder Use(System.Func<Microsoft.AspNetCore.Connections.MultiplexedConnectionDelegate, Microsoft.AspNetCore.Connections.MultiplexedConnectionDelegate> middleware);
+    }
+    public partial interface IMultiplexedConnectionFactory
+    {
+        System.Threading.Tasks.ValueTask<Microsoft.AspNetCore.Connections.MultiplexedConnectionContext> ConnectAsync(System.Net.EndPoint endpoint, Microsoft.AspNetCore.Http.Features.IFeatureCollection features = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken));
+    }
+    public partial interface IMultiplexedConnectionListener : System.IAsyncDisposable
+    {
+        System.Net.EndPoint EndPoint { get; }
+        System.Threading.Tasks.ValueTask<Microsoft.AspNetCore.Connections.MultiplexedConnectionContext> AcceptAsync(Microsoft.AspNetCore.Http.Features.IFeatureCollection features = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken));
```

Comment:
I'm totally on board with BindAsync taking an IFeatureCollection (though an Items `IDictionary<object, object>` might be better) to pass binding metadata (e.g. a TLS cert) that isn't endpoint-specific, but why does AcceptAsync need one? I saw @scalablecory asked for it, but I don't understand the use case.

Reply:
Yeah, I'm going to leave it off for now; we can discuss this more in depth in the future.